sim-cli/src/parsing.rs

```diff
@@ -362,3 +485,17 @@ fn mkdir(dir: PathBuf) -> anyhow::Result<PathBuf> {
     fs::create_dir_all(&dir)?;
     Ok(dir)
 }
+
+pub async fn parse_cli() -> anyhow::Result<(Cli, SimParams)> {
```

Comment:
slight personal preference to call `Cli::parse` at the call site and pass in `&cli` but not strongly held opinion